packages/node-cli/src/commands/object/update/UpdateInputHandler.js

```diff
@@ -54,68 +102,65 @@ module.exports = class UpdateInputHandler extends BaseInputHandler {
 		if (foundXMLFiles.length === 0) {
 			throw NodeTranslationService.getMessage(ERRORS.NO_OBJECTS_IN_PROJECT);
 		}
+		return foundXMLFiles;
+	}
 
-		let filteredObjects;
-
+	async _getObjectsToSelect(foundXMLFiles) {
 		if (foundXMLFiles.length > MAX_ENTRIES_BEFORE_FILTER) {
-			const filterAnswers = await prompt([
-				{
-					type: CommandUtils.INQUIRER_TYPES.LIST,
-					name: ANSWERS_NAMES.FILTER_BY_SCRIPT_ID,
-					message: NodeTranslationService.getMessage(QUESTIONS.FILTER_BY_SCRIPT_ID),
-					default: false,
-					choices: [
-						{ name: NodeTranslationService.getMessage(YES), value: true },
-						{ name: NodeTranslationService.getMessage(NO), value: false },
-					],
-				},
-				{
-					when: (response) => {
-						return response[ANSWERS_NAMES.FILTER_BY_SCRIPT_ID];
-					},
-					type: CommandUtils.INQUIRER_TYPES.INPUT,
-					name: ANSWERS_NAMES.SCRIPT_ID_FILTER,
-					message: NodeTranslationService.getMessage(QUESTIONS.SCRIPT_ID_FILTER),
-					validate: (fieldValue) => showValidationResults(fieldValue, validateScriptId),
-				},
-			]);
-			filteredObjects = filterAnswers[ANSWERS_NAMES.FILTER_BY_SCRIPT_ID]
-				? foundXMLFiles.filter((element) => element.value.includes(filterAnswers[ANSWERS_NAMES.SCRIPT_ID_FILTER]))
-				: foundXMLFiles;
+			const filteredObjects = await this._filterObjectsByScriptId(filteredObjects, foundXMLFiles);
 			if (filteredObjects.length === 0) {
 				throw NodeTranslationService.getMessage(MESSAGES.NO_OBJECTS_WITH_SCRIPT_ID_FILTER);
 			}
+			return filteredObjects;
 		} else {
-			filteredObjects = foundXMLFiles;
+			return foundXMLFiles;
 		}
+	}
+
+	async _filterObjectsByScriptId(filteredObjects, foundXMLFiles) {
```

Comment:
We also don't need here the `fileteredObjects` passed as a parameter, please remove it from the method signature. Later we directly can return the ternary operator result.